core/chains/terra/terratxm/txm.go

```diff
@@ -157,23 +157,45 @@ func unmarshalMsg(msgType string, raw []byte) (sdk.Msg, string, error) {
 	return nil, "", errors.Errorf("unrecognized message type: %s", msgType)
 }
 
+type msgValidator struct {
+	cutoff         time.Time
+	expired, valid terra.Msgs
+}
+
+func (e *msgValidator) add(msg terra.Msg) {
+	if msg.CreatedAt.Before(e.cutoff) {
+		e.expired = append(e.expired, msg)
+	} else {
+		e.valid = append(e.valid, msg)
+	}
+}
+
 func (txm *Txm) sendMsgBatch(ctx context.Context) {
-	var notExpired, expired terra.Msgs
+	msgs := msgValidator{cutoff: time.Now().Add(-txm.cfg.TxMsgTimeout())}
 	err := txm.orm.q.Transaction(func(tx pg.Queryer) error {
-		unstarted, err := txm.orm.GetMsgsState(db.Unstarted, txm.cfg.MaxMsgsPerBatch(), pg.WithQueryer(tx))
+		started, err := txm.orm.GetMsgsState(db.Started, txm.cfg.MaxMsgsPerBatch(), pg.WithQueryer(tx))
```

Comment:
this is in case we crash after marking started but before broadcasting right? Worth a comment